### PR TITLE
Terminology configuration validation

### PR DESCRIPTION
**Pull request by nickrflynn**
Validate required services and databases before configuration
Validate Terminology service and database after configuration

### CODE REVIEW

File: Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1 [hunk 1]
```diff
@@ -52,33 +52,110 @@ function Get-ConfigValue {
     }
 }
 
-function Invoke-SqlCommand{
+function Get-ServiceFromDiscovery {
+    param(
+        [String] $DiscoveryUrl = $(throw "Please specify the Discovery Service Uri"),
```
Comment (benjanderson) [inline Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1]: I'm ok with this, but couldn't you just use `[ValidateNotNullorEmpty()]`

File: Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1 [hunk 2]
```diff
@@ -52,33 +52,110 @@ function Get-ConfigValue {
...
+    )
+
+    $discoveryServiceQuery = "/Services?`$filter=ServiceName eq '$Name' and Version eq $Version&`$select=ServiceUrl"
+
+    $uri = "$DiscoveryUrl$discoveryServiceQuery"
+
+    try {
+        $response = Invoke-WebRequest -Uri $uri -Method GET -UseDefaultCredentials
+    }
+    catch {
```
Comment (benjanderson) [inline Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1]: I think you may want to specify the type in this catch. If something else were to go wrong, this exception would be very obscure

File: Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1 [hunk 3]
```diff
@@ -52,33 +52,110 @@ function Get-ConfigValue {
...
+    )
+    
+    $uriWithoutVersion = $ServiceUrl.Substring(0, $ServiceUrl.LastIndexOf('/'))
+    $uri = "$uriWithoutVersion/Ping"
+
+    try {
+        Write-DosMessage -Level "Information" -Message "Attempting to ping $ServiceName ($uri)"  
+        Invoke-WebRequest -Uri $uri -Method GET -UseDefaultCredentials | Out-Null
+    }
+    catch {
```
Comment (benjanderson) [inline Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1]: Here too

File: Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1 (hunk 1 above)
Comment (nickrflynn) [reply Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1]: I can (and probably should).  I saw that used in Imran's email but I had already been validating this way and forgot.  `[ValidateNotNullorEmpty()]` would be better.  I'll change it.

Comment (benjanderson) [reply Fabric.Terminology.API/Scripts/Terminology-Install-Utilities.psm1]: I agree with validating parameters, but there is a cleaner way

**Pull request merged**